File: elfa/a3_fast_api/app/api/health.py
from datetime import datetime
from fastapi import APIRouter
from pydantic import BaseModel

from app.core.config import settings


class HealthResponse(BaseModel):
    status: str
    app: str
    env: str
    timestamp: datetime


router = APIRouter()


@router.get("/", response_model=HealthResponse)
async def health_check():
    """Endpoint de healthcheck da aplicação"""
    return HealthResponse(
        status="healthy",
        app=settings.app_name,
        env=settings.app_env,
        timestamp=datetime.now()
    )

File: elfa/elfa_backend/src/api/health.js
import express from 'express';
import { config } from '../core/config.js';

const router = express.Router();

/**
 * @swagger
 * /health:
 *   get:
 *     summary: Verifica o status de saúde da aplicação
 *     tags: [Health]
 *     description: Retorna o status atual da aplicação, ambiente e timestamp
 *     responses:
 *       200:
 *         description: Aplicação está saudável
 *         content:
 *           application/json:
 *             schema:
 *               type: object
 *               properties:
 *                 status:
 *                   type: string
 *                   example: healthy
 *                 app:
 *                   type: string
 *                   example: elfa-backend
 *                 env:
 *                   type: string
 *                   example: local
 *                 timestamp:
 *                   type: string
 *                   format: date-time
 *                   example: 2024-01-29T10:30:00.000Z
 */
router.get('/', (req, res) => {
  res.json({
    status: 'healthy',
    app: config.appName,
    env: config.env,
    timestamp: new Date().toISOString()
  });
});

export default router;
